tktermwidget/style.py

```diff
@@ -1,27 +1,275 @@
+    foregroundentry = Entry(foregroundframe)
+    foregroundbutton = Button(
+        foregroundframe, text="...", width=3, command=lambda: selectcolor(foregroundentry, "foreground")
+    )
+
+    buttonframe = Frame(config)
+    save = Button(buttonframe, text="Save", command=savestyle)
+    cancel = Button(buttonframe, text="Cancel", command=config.destroy)
+
+    # TODO: Add buttons and labels to help user to config the style
+    render = Text(
+        config,
+        width=40,
+        background=style["background"],
+        insertbackground=style["insertbackground"],
+        selectbackground=style["selectbackground"],
+        selectforeground=style["selectforeground"],
+        foreground=style["foreground"],
+        font=("Cascadia Mono", 9, "normal"),
+        relief="flat",
+    )
+
+    # TODO: Improve here
+    render.insert("insert", "This is a normal text for test style.")
+    render.tag_add("select", "1.31", "1.36")
+    render.tag_config("select", background=style["selectbackground"], foreground=style["selectforeground"])
+    render["state"] = "disable"
+
+    if theme:
+        for widget in (
+            backgroundbutton,
+            insertbackgroundbutton,
+            selectbackgroundbutton,
+            selectforegroundbutton,
+            foregroundbutton,
+            save,
+        ):
+            widget.config(style="Accent.TButton")
+
+    save.pack(side="right")
+    cancel.pack(side="right")
+    buttonframe.pack(side="bottom", fill="x")
+
+    render.pack(side="right", fill="y", padx=3, pady=3)
+    create.pack(side="top", fill="y", pady=15)
+
+    for widget in (
+        background,
+        insertbackground,
+        selectbackground,
+        selectforeground,
+        foreground,
+        backgroundentry,
+        insertbackgroundentry,
+        selectbackgroundentry,
+        selectforegroundentry,
+        foregroundentry,
+        backgroundbutton,
+        insertbackgroundbutton,
+        selectbackgroundbutton,
+        selectforegroundbutton,
+        foregroundbutton,
+    ):
+        widget.pack(side="left", padx=3)
+
+    for widget in (backgroundframe, insertbackgroundframe, selectbackgroundframe, selectforegroundframe, foregroundframe):
+        widget.pack(side="top", fill="y")
+
+    config.mainloop()
+
+
+# Styles format:
+# {yourstylename}: dict[str, str, str, str, str] = {
+#    "background": "{yourhexcolor}",
+#    "insertbackground": "{yourhexcolor}",
+#    "selectbackground": "{yourhexcolor}",
+#    "selectforeground": "{yourhexcolor}",
+#    "foreground": "{yourhexcolor}",
+# }
+
+# Built-in styles
+DEFAULT: dict[str, str, str, str, str] = {  # Style for normal tkterminalwidget
     "background": "#2B2B2B",
     "insertbackground": "#DCDCDC",
     "selectbackground": "#b4b3b3",
     "selectforeground": "#e6e6e6",
     "foreground": "#cccccc",
 }
 
-Powershell: dict = {  # Style for powershell
+POWERSHELL: dict[str, str, str, str, str] = {  # Style for powershell
```

Comment:
Just use dict[str]. The repetition isn't quite necessary ;).

Reply:
Thanks for you tip! (I never wrote it before)